File: src/main/java/cn/labzen/meta/exception/LabzenException.java
package cn.labzen.meta.exception;

import org.slf4j.helpers.MessageFormatter;

public class LabzenException extends Exception {

  public LabzenException(String message) {
    super(message);
  }

  public LabzenException(String message, String... args) {
    super(MessageFormatter.basicArrayFormat(message, args));
  }

  public LabzenException(Throwable cause) {
    super(cause);
  }

  public LabzenException(Throwable cause, String message) {
    super(message, cause);
  }

  public LabzenException(Throwable cause, String message, String... args) {
    super(MessageFormatter.basicArrayFormat(message, args), cause);
  }
}

File: src/main/kotlin/cn/labzen/meta/exception/LabzenException.kt
package cn.labzen.meta.exception

import org.slf4j.helpers.MessageFormatter

abstract class LabzenException : Exception {
  constructor(message: String) : super(message)
  constructor(message: String, vararg arguments: Any?) : super(MessageFormatter.basicArrayFormat(message, arguments))

  constructor(cause: Throwable) : super(cause)
  constructor(cause: Throwable, message: String) : super(message, cause)
  constructor(cause: Throwable, message: String, vararg arguments: Any?) :
      super(MessageFormatter.basicArrayFormat(message, arguments), cause)
}
